src/components/policy/src/policy/include/policy/policy_manager_impl.h

```diff
@@ -187,6 +188,25 @@ class PolicyManagerImpl : public PolicyManager {
 
     virtual std::string RetrieveCertificate() const OVERRIDE;
 
+    class RetrySequenceTimewatch {
+     public:
+      RetrySequenceTimewatch(PolicyManagerImpl* ptr);
+      ~RetrySequenceTimewatch();
+      void Stop();
+      bool IsRunning();
+      void Start(uint32_t timeout);
+
+     private:
+      typedef timer::TimerThread<RetrySequenceTimewatch> Timer;
+      Timer timer_retry_sequence_;
+
+      PolicyManagerImpl* policy_manager_;
+
+     private:
+      RetrySequenceTimewatch();
```

Comment:
If create this object without parameter will be UB. So, I want, that possible users of my class couldn't create object of my class without parameter, so if they want use without parameter they have to modify class.